pkg/config/config.go

```diff
@@ -21,45 +21,46 @@ const defaultConfigFile = "athens.toml"
 // Config provides configuration values for all components
 type Config struct {
 	TimeoutConf
-	GoEnv             string    `validate:"required" envconfig:"GO_ENV"`
-	GoBinary          string    `validate:"required" envconfig:"GO_BINARY_PATH"`
-	GoProxy           string    `envconfig:"GOPROXY"`
-	GoBinaryEnvVars   EnvList   `envconfig:"ATHENS_GO_BINARY_ENV_VARS"`
-	GoGetWorkers      int       `validate:"required" envconfig:"ATHENS_GOGET_WORKERS"`
-	GoGetDir          string    `envconfig:"ATHENS_GOGOET_DIR"`
-	ProtocolWorkers   int       `validate:"required" envconfig:"ATHENS_PROTOCOL_WORKERS"`
-	LogLevel          string    `validate:"required" envconfig:"ATHENS_LOG_LEVEL"`
-	CloudRuntime      string    `validate:"required" envconfig:"ATHENS_CLOUD_RUNTIME"`
-	EnablePprof       bool      `envconfig:"ATHENS_ENABLE_PPROF"`
-	PprofPort         string    `envconfig:"ATHENS_PPROF_PORT"`
-	FilterFile        string    `envconfig:"ATHENS_FILTER_FILE"`
-	TraceExporterURL  string    `envconfig:"ATHENS_TRACE_EXPORTER_URL"`
-	TraceExporter     string    `envconfig:"ATHENS_TRACE_EXPORTER"`
-	StatsExporter     string    `envconfig:"ATHENS_STATS_EXPORTER"`
-	StorageType       string    `validate:"required" envconfig:"ATHENS_STORAGE_TYPE"`
-	GlobalEndpoint    string    `envconfig:"ATHENS_GLOBAL_ENDPOINT"` // This feature is not yet implemented
-	Port              string    `envconfig:"ATHENS_PORT"`
-	BasicAuthUser     string    `envconfig:"BASIC_AUTH_USER"`
-	BasicAuthPass     string    `envconfig:"BASIC_AUTH_PASS"`
-	PropagateAuthHost string    `envconfig:"ATHENS_PROPAGATE_AUTH_HOST"`
-	ForceSSL          bool      `envconfig:"PROXY_FORCE_SSL"`
-	ValidatorHook     string    `envconfig:"ATHENS_PROXY_VALIDATOR"`
-	PathPrefix        string    `envconfig:"ATHENS_PATH_PREFIX"`
-	NETRCPath         string    `envconfig:"ATHENS_NETRC_PATH"`
-	GithubToken       string    `envconfig:"ATHENS_GITHUB_TOKEN"`
-	HGRCPath          string    `envconfig:"ATHENS_HGRC_PATH"`
-	TLSCertFile       string    `envconfig:"ATHENS_TLSCERT_FILE"`
-	TLSKeyFile        string    `envconfig:"ATHENS_TLSKEY_FILE"`
-	SumDBs            []string  `envconfig:"ATHENS_SUM_DBS"`
-	NoSumPatterns     []string  `envconfig:"ATHENS_GONOSUM_PATTERNS"`
-	DownloadMode      mode.Mode `envconfig:"ATHENS_DOWNLOAD_MODE"`
-	DownloadURL       string    `envconfig:"ATHENS_DOWNLOAD_URL"`
-	SingleFlightType  string    `envconfig:"ATHENS_SINGLE_FLIGHT_TYPE"`
-	RobotsFile        string    `envconfig:"ATHENS_ROBOTS_FILE"`
-	IndexType         string    `envconfig:"ATHENS_INDEX_TYPE"`
-	SingleFlight      *SingleFlight
-	Storage           *Storage
-	Index             *Index
+	GoEnv             string        `validate:"required" envconfig:"GO_ENV" json:"goenv"`
```

Comment:
I don't think we need to add any `json:""` tags in this struct. We can just keep the outputted JSON with upper cases. It's quite common for example the GOPROXY itself outputs is `.info` file with upper case fields: https://proxy.golang.org/github.com/pkg/errors/@v/v0.9.1.info